neo/rawio/intanrawio.py

```diff
@@ -148,7 +228,12 @@ def _get_signal_size(self, block_index, seg_index, stream_index):
         signal_channels = self.header["signal_channels"][mask]
         channel_names = signal_channels["name"]
         chan_name0 = channel_names[0]
-        size = self._raw_data[chan_name0].size
+        if self.file_format == "header-attached":
+            size = self._raw_data[chan_name0].size
+        elif self.file_format == 'one-file-per-signal':
+            size = self._raw_data[stream_index][:,0].size
```

Comment:
Same here, I was trying to think about whether this was the correct way. I think this one is okay, but want to read this carefully.

Reply:
This function is mean to calculate the number of samples, right? For this file_format case you already loaded the memmap with a specific shape:

https://github.com/zm711/python-neo/blob/839060e5b910d878021348de34f50e7bf1ce343e/neo/rawio/intanrawio.py#L112-L120

So you know what the number of samples is. This will avoid you accessing the data of the memmap as you are currently doing which I think is bad because it loads  data into memory.

Reply:
Great point. I wasn't even thinking about that.

Reply:
I mean just getting the first element of shape should do the deed, right?

Reply:
Fixed. Now that I understand what you mean :)